docs/source/telegram.ext.rst

```diff
@@ -53,11 +53,3 @@ Arbitrary Callback Data
 
     telegram.ext.callbackdatacache
     telegram.ext.invalidcallbackdata
-
-utils
------
-
-.. toctree::
-
-    telegram.ext.utils.promise
-    telegram.ext.utils.types
```

Comment:
What benefit do you see in including modules in the documentation that are explicitly marked as private? As long as they were not private, it made sense and I added the warnings more or less so that we can say "I told you so" in case someone uses them and we break it. But with the utility modules being private now, I don't see a reason to include them in public docs.

The point about type aliases is valid though - thanks for catching this! Some solutions that I could image are:

1. rework documentation without mentioning the type aliases modules
2. make the `types` modules public, i.e. move `telgram(.ext)._utils.types` to `telegram(.ext).types` and remove the warning - then changes have to be considered breaking
3. Midway between the current state of v14 and 2.: Move the modules to `telegram(.ext).types` for convenience of the user, but keep the warning.
4. Another possible midway: The type aliases `CDCData` and `ConversationDict` might actually be of some use for advanced users that build their own persistence classes. We could add them as class variables to `CallbackDataCache` and `ConversationHandler` respectively, similarly to what I've done with `CallbackContext.DEFAULT_TYPE` in #2646. The rest is of no real interest for users IMO, so we could hide them in private modules and adapt the parts of the docs that mention them.

What do you think?

Reply:
> What benefit do you see in including modules in the documentation that are explicitly marked as private?

I was thinking that this could also lower the entry bar for new contributors since 'everything' is documented for them. But maybe it isn't that important since they can dig in the source code itself. Might also reduce new users from being overwhelmed by the amount of documentation.

Also I'd like to propose a 5th point:
- Keep it the way it is now. Meaning make `telegram(.ext).utils` a public module, but keep certain files which aren't meant to be used by users at all private and hidden from documentation. Like how `webhookhandler.py` is not included in the docs of  `telegram.ext.utils` at all. The directory structure would be like:

```
telegram.ext.utils:
    telegram.ext.utils.promise.Promise
    telegram.ext.utils.types Module

telegram.utils:
    telegram.utils.defaultvalue Module
    telegram.utils.types Module
```
The remaining files would be in `utils` but would be private and not in documentation (since they aren't being referenced anywhere else). And we keep the warning in all of them.

Reply:
@kennethcheo 

Harshil & 1 had a chat offline and we came to the conclusion that all contents of `(ext)._utils` should stay private. This means that some parts of the documentation need to be updated, more precisely the docs shouldn't mention any contents of `(ext)._utils`. I'm aware that this wasn't part of the description in #2680. Still, if you would like to try & make the corresponding changes, you're very welcome to do so. If you don't want to, that's ofc fine as well.
I would be glad if you could give a short response here 🙂